templates/http-go/metadata/manifest.toml

```diff
@@ -0,0 +1,9 @@
+template_version = "1"
```

Comment:
I think we need to somehow indicate that this is a "_Spin_ template manifest". Some options, in lann-preference-order:

1. rename the `metadata` dir to something like `spin-template`, maybe even `.spin-template`
1. rename `manifest.toml` to `spin-template.toml`
1. change this key to `spin_template_version`

Reply:
Hah, I was thinking we should change `manifest.toml` to `spin-template.toml` or some such!

I was also thinking we might want to let authors version their templates (for diagnostic purposes).  This field is really the "schema version" - maybe there should be a true "template version" field as well...

Reply:
Aha! I'm fine with no explicit schema version - you can always treat its absence as version "0" if you end up needing it.

Reply:
* Changed file name to `spin-template.toml`
* Changed field name to `manifest_version`

Tentatively resolving this but happy to keep iterating!